source/extensions/tracers/xray/localized_sampling.h

```diff
@@ -120,13 +120,18 @@ class LocalizedSamplingManifest {
    */
   LocalizedSamplingRule& defaultRule() { return default_rule_; }
 
+  /**
+   * @return default sampling rule's sampling rate. Mainly for unit testing purposes.
+   */
+  double defaultRuleRate() const { return default_rule_.rate(); }
+
```

Comment:
Is this necessary? Why can't you just call `defaultRule().rate()`?

Reply:
From earlier suggestion in this PR LocalizedSamplingStrategy::defaultManifest() function was made a const. 
`const LocalizedSamplingManifest& defaultManifest() const`.

So the return type is a **const** reference of LocalizedSamplingManifest object. To make a call `strategy.defaultManifest().defaultRule().rate()` in the UnitTest code, the function `LocalizedSamplingManifest::defaultRule()` was not expected to be called by a const reference of LocalizedSamplingManifest object.

Instead of making defaultRule function as const `LocalizedSamplingManifest::defaultRule const()` which in turn needed multiple changes where defaultRule function is called within the source code, I opted to crating a new function `double defaultRuleRate() const {}` which seemed like a less code change.

Reply:
Can you not just make a copy of the default rule in the unit test from the cost ref?

Reply:
> Is this necessary? Why can't you just call `defaultRule().rate()`?

It is not necessary. In the new commit, got rid of this function (`defaultRuleRate`) and instead making a copy of the default_manifest_ from the const ref